File: dataset/human/java/file_9759.java
import java.lang.Math.max

data class Node(var parentNodeNum: Int = -1, var sibling: Int = -1, var degree: Int = 0, var depth: Int = 0, var height: Int = 0,
                var nodeType: NodeType = NodeType.ROOT, var left: Pair<Int, Node>? = null, var right: Pair<Int, Node>? = null)
enum class NodeType {
    ROOT {
        override fun toString() = "root"
    },
    INTERNAL_NODE {
        override fun toString() = "internal node"
    },
    LEAF {
        override fun toString() = "leaf"
    }
}

fun main(args: Array<String>) {
    val inputList = List(readLine()?.toInt() ?: return) {
        readLine()?.split(" ".toRegex())?.map(String::toInt) ?: listOf()
    }
    val nodeMap = mutableMapOf<Int, Node>()
    inputList.forEach { input ->
        val nodeNum = input[0]
        nodeMap.putIfAbsent(nodeNum, Node())
        nodeMap[nodeNum]?.let { node ->
            if (input[1] != -1) {
                val leftNum = input[1]
                nodeMap.putIfAbsent(leftNum, Node())
                node.left = leftNum to nodeMap[leftNum]!!
                nodeMap[leftNum]?.parentNodeNum = nodeNum
                node.degree++
            }
            if (input[2] != -1) {
                val rightNum = input[2]
                nodeMap.putIfAbsent(rightNum, Node())
                node.right = rightNum to nodeMap[rightNum]!!
                nodeMap[rightNum]?.parentNodeNum = nodeNum
                node.degree++
            }
            node.left?.second?.sibling = node.right?.first ?: -1
            node.right?.second?.sibling = node.left?.first ?: -1
        }
    }
    recDepth(nodeMap.values.first { it.parentNodeNum == -1 }, 0)
    val stringBuilder = StringBuilder()
    for (i in inputList.indices) {
        nodeMap[i]?.let { node ->
            stringBuilder.append("node $i: parent = ${node.parentNodeNum}, sibling = ${node.sibling}, degree = ${node.degree}, depth = ${node.depth}, height = ${node.height}, ${node.nodeType}\n")
        }
    }
    print(stringBuilder.toString())
}

fun recDepth(node: Node, depth: Int): Int {
    node.depth = depth
    when {
        node.parentNodeNum == -1 -> node.nodeType = NodeType.ROOT
        node.right == null && node.left == null -> node.nodeType = NodeType.LEAF
        else -> node.nodeType = NodeType.INTERNAL_NODE
    }
    val leftHeight = if (node.left != null) {
        recDepth(node.left!!.second, depth + 1)
    } else {
        0
    }
    val rightHeight = if (node.right != null) {
        recDepth(node.right!!.second, depth + 1)
    } else {
        0
    }
    node.height = max(leftHeight, rightHeight)
    return node.height + 1
}

File: dataset/human/c/file_9759.c
#include <stdio.h>

#define MAX 100005
#define R -1

typedef struct{
  int parent;
  int right; //??????
  int left;  //??????
}Node;

Node A[MAX];
int n;
int B[MAX];
int C[MAX];

void depth(int u,int p){
  B[u]=p;
  if(A[u].right!=R) depth(A[u].right,p);
  if(A[u].left!=R) depth(A[u].left,p+1);
}
void print2(int i){
  int c,j;
  c=A[i].left;
  for(j=0,c=A[i].left;c!=R;j++,c=A[c].right){
    if(j)printf(", ");
    printf("%d",c);
  }
}

void print(){
  int i;
  for(i=0;i<n;i++){
    printf("node %d: ",i);
    printf("parent = %d, ",A[i].parent);
    printf("depth = %d, ",B[i]);
    if(A[i].parent==R) printf("root, ");
    else if(A[i].left==R) printf("leaf, ");
    else printf("internal node, ");
    printf("[");
    print2(i);
    printf("]\n");
  }
}

int main(int argc, char const *argv[]) {
  int i,j,root,cnt,r=0,c,x,s;
  scanf("%d",&n);

  for(i=0;i<n;i++){
    A[i].parent=R;
    A[i].left=R;
    A[i].right=R;
  }
  for(i=0;i<n;i++){
    scanf("%d",&root);
    scanf("%d",&C[i]);
    for(j=0;j<C[i];j++){
      scanf("%d",&c);
      A[c].parent=root;
      if(j==0) A[root].left=c;
      else A[x].right=c;
      x=c;
    }
  }
  for(i=0;i<n;i++){
    if(A[i].parent==R) s=i;
  }
  depth(s,0);
  print();
  return 0;
}
